plugin/plaintextmodel.cpp

```diff
@@ -264,6 +277,65 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detectCodec(const char *encodedCharacters, qint64 length)
+{
+    QLocale locale;
+    QTextCodec *codec = QTextCodec::codecForName("UTF-8");
+
+    UErrorCode error = U_ZERO_ERROR;
+    if (UCharsetDetector * const detector = ucsdet_open(&error)) {
+        ucsdet_setText(detector, encodedCharacters, length, &error);
+
+        int32_t count = 0;
+        if (U_FAILURE(error)) {
+            qWarning() << "Unable to detect text encoding" << u_errorName(error);
+        } else if (const UCharsetMatch ** const matches = ucsdet_detectAll(detector, &count, &error)) {
+            int bestConfidenceICU = 0;
+            QTextCodec * bestMatchICU = nullptr;
+
+            for (int32_t i = 0; i < count; ++i) {
+                if (QTextCodec *detectedCodec = QTextCodec::codecForName(ucsdet_getName(matches[i], &error))) {
+                    bestConfidenceICU = ucsdet_getConfidence(matches[i], &error);
+                    bestMatchICU = detectedCodec;
+
+                    break;
+                }
+            }
+
+            if(bestMatchICU) {
+                QString localeName = locale.name();
+
+                int bestConfidenceLan = 0;
+                QTextCodec * bestMatchLan = nullptr;
+
+                for (int32_t i = 0; i < count; ++i) {
+                    if (localeName.left(localeName.indexOf("_")) == QString(ucsdet_getLanguage(matches[i], &error))) {
+                        if(QTextCodec * detectedCodec = QTextCodec::codecForName(ucsdet_getName(matches[i], &error))) {
+                            bestConfidenceLan = ucsdet_getConfidence(matches[i], &error) * 1.5;
+                            bestMatchLan = detectedCodec;
+
+                            break;
+                        }
+                    }
+                }
+
+                if (bestMatchLan) {
+                    bestConfidenceICU > bestConfidenceLan ? codec = bestMatchICU : codec = bestMatchLan;
```

Comment:
Assignments in the operands is uncommon and also unnecessary. Could be just codec = bestConfidence_etc ? bestMatchICU : bestMatchLan;

Reply:
Ok. I'll correct

Reply:
Done